data/tests/tests_framework.txt

```diff
@@ -50,3 +50,181 @@ test "Test-Framework - Conditions Arithmetic and Loops"
 			10 == "test: loopControl"
 			"test: myFirstTestVariable" == 200 - 50
 			400 * 1024 == "test: mySecondTestVariable"
+
+
+test "Test-Framework - UI Keyboard controls"
+	status active
+	description "Tests if the in-game panels can be controlled by keypresses"
+	sequence
+		# Create/inject the savegame, go to the load screen and load the first (and only) savegame.
+		inject "Three Earthly Barges"
+		input
+			key l
+		input
+			key l
+		# Enter and leave jobs
+		input
+			key j
+		input
+			key d
+		# Enter trading
+		input
+			key t
+		# Enter bank
+		input
+			key b
+		# Enter spaceport
+		input
+			key p
+		# Enter and leave shipyard
+		# Skipped, since Earth (the starting planet) doesn't have a shipyard.
+		#input
+		#	key s
+		#input
+		#	key l
+		# Enter and leave outfitter
+		input
+			key o
+		input
+			key l
+		# Enter hire-crew section
+		input
+			key h
+		# Depart
+		input
+			key d
+
+
+test "Test-Framework - UI Pointer controls"
+	status "missing feature"
+	description "Tests if the in-game panels can be controlled by pointer controls"
+	sequence
+		# Create/inject the savegame, go to the load screen and load the first (and only) savegame.
+		inject "Three Earthly Barges"
+		input
+			key l
+		input
+			key l
+		# Enter and leave jobs
+		input
+			key j
+		input
+			key d
+		# Enter trading
+		input
+			pointer
+				# Go from middle of the screen (to allow many resolutions)
+				X 0.5 relative
+				Y 0.5 relative
+				# Go for the center of the trading button
+				X -411
+				Y 92
+				click left
```

Comment:
How robust is this going to be to e.g. interfaces.txt changes? If a plugin shifts things around, will all of its users be unable to use automated tests without seeing these fail?

Reply:
The ratio of "the risk of the test breaking" vs "the benefit of testing mouse-clicks here" is indeed not favorable for this test. This part of the game is better tested using key-presses only. I'll remove this test.

We could increase the stability of the tests a little by giving tests their own data(sets) files, but in general it is possible that changes to the game also require changes to the relevant tests (tests test against current expected behavior, which can be different from future expected behavior).